build fail looks real
```
======================================================================
FAIL: test_get_users_assigned_to_locations (corehq.apps.locations.tests.test_dbaccessors.TestUsersByLocation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/locations/tests/test_dbaccessors.py", line 79, in test_get_users_assigned_to_locations
    [self.varys._id, self.tyrion._id, self.daenerys._id, self.george._id]
AssertionError: Element counts were not equal:
First has 0, Second has 1:  u'e1a70f879d8f0166fb1e63e40f99d466'
First has 0, Second has 1:  u'e1a70f879d8f0166fb1e63e40f99d3a0'
----------------------------------------------------------------------
```

--- a/corehq/apps/locations/tests/test_permissions.py
+++ b/corehq/apps/locations/tests/test_permissions.py
@@ -2,7 +2,9 @@ import random
 import string
 
 from corehq.apps.users.models import WebUser, CommCareUser
-from corehq.toggles import RESTRICT_FORM_EDIT_BY_LOCATION, NAMESPACE_DOMAIN
+from corehq.toggles import (RESTRICT_FORM_EDIT_BY_LOCATION, NAMESPACE_DOMAIN,
+                            MULTIPLE_LOCATIONS_PER_USER)
+from corehq.util.test_utils import flag_enabled
 from couchforms.models import XFormInstance
 
 from ..permissions import can_edit_form_location
@@ -46,6 +48,17 @@ class TestPermissions(LocationHierarchyTestCase):
         locationless_web_user = WebUser.create(self.domain, 'joeshmoe', 'password')
         self.assertCannotEdit(locationless_web_user, self.cambridge_form)
         self.assertCannotEdit(locationless_web_user, self.boston_form)
+
+    @flag_enabled(MULTIPLE_LOCATIONS_PER_USER)
+    def test_multiple_locations_per_user(self):
+        # Note also that location types must not be administrative for multiple
+        # locations per domain to work.  This was a pain to figure out...
+        multi_loc_user = self.make_mobile_user('Cambridge')
+        multi_loc_user.set_location(self.locations['Boston'].couch_location)
+        multi_loc_form = self.make_form(multi_loc_user)
+
+        self.assertCanEdit(self.middlesex_web_user, multi_loc_form)
+        self.assertCanEdit(self.massachusetts_web_user, multi_loc_form)
 
     #### The rest of this class is helper methods and setup ####
 

--- a/corehq/apps/locations/tests/util.py
+++ b/corehq/apps/locations/tests/util.py
@@ -37,7 +37,7 @@ def _setup_location_types(domain, location_types):
             domain=domain,
             name=name,
             parent_type=previous,
-            administrative=True,
+            administrative=False,
         )
         location_types_dict[name] = previous = location_type
     return location_types_dict

--- a/corehq/apps/users/models.py
+++ b/corehq/apps/users/models.py
@@ -1850,14 +1850,17 @@ class CommCareUser(CouchUser, SingleMembershipMixin, CommCareMobileContactMixin)
         self.save()
 
     @property
-    def _locations(self):
+    def locations(self):
         """
-        Hidden access to a users delgate location list.
+        This method is only used for domains with the multiple
+        locations per user flag set. It will error if you try
+        to call it on a normal domain.
+        """
+        if not self.project.supports_multiple_locations_per_user:
+            raise InvalidLocationConfig(
+                "Attempting to access multiple locations for a user in a domain that does not support this."
+            )
 
-        Should be removed (and this code moved back under
-        self.location) sometime after migration and things
-        are settled.
-        """
         from corehq.apps.locations.models import Location
         from corehq.apps.commtrack.models import SupplyPointCase
 
@@ -1880,20 +1883,6 @@ class CommCareUser(CouchUser, SingleMembershipMixin, CommCareMobileContactMixin)
                 yield Location.wrap(doc)
 
         return list(_gen())
-
-    @property
-    def locations(self):
-        """
-        This method is only used for domains with the multiple
-        locations per user flag set. It will error if you try
-        to call it on a normal domain.
-        """
-        if not self.project.supports_multiple_locations_per_user:
-            raise InvalidLocationConfig(
-                "Attempting to access multiple locations for a user in a domain that does not support this."
-            )
-
-        return self._locations
 
     def supply_point_index_mapping(self, supply_point, clear=False):
         from corehq.apps.commtrack.exceptions import (

--- a/corehq/util/test_utils.py
+++ b/corehq/util/test_utils.py
@@ -1,4 +1,5 @@
 import json
+import mock
 import os
 
 from functools import wraps
@@ -54,3 +55,17 @@ class TestFileMixin(object):
     @classmethod
     def get_xml(cls, name, override_path=None):
         return cls.get_file(name, 'xml', override_path)
+
+
+def flag_enabled(toggle_class):
+    """
+    Decorate test methods with this to mock the lookup
+
+        @flag_enabled(toggles.MULTIPLE_LOCATIONS_PER_USER)
+        def test_something_fancy(self):
+            something.which_depends(on.MULTIPLE_LOCATIONS_PER_USER)
+    """
+    return mock.patch(
+        '.'.join([toggle_class.__module__, toggle_class.__class__.__name__, 'enabled']),
+        new=lambda *args: True,
+    )

